Add ocl-icd-system shim package

jaimergp opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`
- ruby `@conda-forge/help-ruby`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams.
Consider asking on our [Gitter channel](https://gitter.im/conda-forge/conda-forge.github.io)
if your recipe isn't reviewed promptly.
-->

Following conversations had in https://github.com/conda-forge/openmm-feedstock/pull/3#issuecomment-540083831 and https://github.com/openmm/openmm/issues/2401, this shim package will create the needed symlinks to make system OpenCL installations visible to `ocl-icd`. Hopefully this is a good compromise for everyone involved!

I am not sure about the name, so please feel free to suggest better options. Also, what license applies here?

__Checklist__

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vend other packages
- [X] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/ocl-icd-system```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipes/ocl-icd-system**:

 * license_file entry is missing, but is expected.

isuruf commented on recipes/ocl-icd-system/activate.sh:
```diff
@@ -0,0 +1,12 @@
+    if [[ -z "$(ls ${CONDA_PREFIX}/etc/OpenCL/vendors/)" ]]; then
+        # and a system installation is available
+        if [[ ! -z "$(ls /etc/OpenCL/vendors/*.icd 2> /dev/null || true)" ]]; then
+            ln -s /etc/OpenCL/vendors "${CONDA_PREFIX}"/etc/OpenCL/vendors/system
```
I would create this symlink irrespective of whether there's a system installation or not.

isuruf commented on recipes/ocl-icd-system/activate.sh:
```diff
@@ -0,0 +1,9 @@
+    # If no OpenCL ICD has been installed with conda yet
+    if [[ -z "$(ls ${CONDA_PREFIX}/etc/OpenCL/vendors/)" ]]; then
+        ln -s /etc/OpenCL/vendors "${CONDA_PREFIX}"/etc/OpenCL/vendors/system
+        touch "${CONDA_PREFIX}"/etc/OpenCL/vendors/.ocl_icd_system_cf
```
Why this ?

jaimergp replied:
To make sure it was created by this package, and not by the user. It would be very unlucky, but possible?

isuruf replied:
Why not do change the name from `system` to `system_cf` or something similar.

jaimergp replied:
What about `ocl-icd-system` so it is consistent with the package name?

isuruf replied:
Sure.

isuruf commented on recipes/ocl-icd-system/activate.sh:
```diff
@@ -0,0 +1,9 @@
+conda_ocl_icd_system_activate () {
+    # If no OpenCL ICD has been installed with conda yet
+    if [[ -z "$(ls ${CONDA_PREFIX}/etc/OpenCL/vendors/)" ]]; then
```
Same as before. No need to check this.

isuruf commented on recipes/ocl-icd-system/activate.sh:
```diff
@@ -0,0 +1,9 @@
+conda_ocl_icd_system_activate () {
```
Can you make this a `post-link.sh` script and the deactivate a `pre-unlink.sh` script.

jaimergp commented:
That last commit addresses all suggestions. If we all agree on the name and descriptions in `meta.yaml`, I am happy with the result. Also, do I need to include a license?

isuruf commented:
You can add https://github.com/conda-forge/staged-recipes/blob/master/LICENSE.txt as a license.

jakirkham commented:
cc @peastman
